firstapp/app/src/androidTest/java/com/example/firstapp/UtilityMethods.java

```diff
@@ -1,20 +1,30 @@
 package com.example.firstapp;
 import android.content.Context;
 import android.content.Intent;
+import android.os.RemoteException;
+import android.os.SystemClock;
+
 import androidx.test.core.app.ApplicationProvider;
 import androidx.test.uiautomator.By;
 import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
 import androidx.test.uiautomator.Until;
-import androidx.test.platform.app.InstrumentationRegistry;
 import org.hamcrest.CoreMatchers;
 import org.junit.Assert;
 
+import java.io.IOException;
+
+import static java.lang.Thread.sleep;
 
-public class UtilityMethods {
-    static UiDevice launchApp(String pkg, int timeout) {
-        // Initialize UiDevice instance
-        UiDevice device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
 
+public class UtilityMethods {
+    /*
+    Launches the specified App on the specified device and returns the time just before the app was
+    launched--to be used later for timing startup.
+     */
+    static long launchApp(UiDevice device, String pkg, int timeout) throws InterruptedException {
```

Comment:
1. static utility methods should generally be public, unless there is a specific reason not to make it public
2. timeout -> timeoutMs?